src/csm_testing/tests/sat_functional/bootprep/test_bootprep.py

```diff
@@ -439,36 +428,84 @@ def delete_cfs_configuration(cfs_config_name):
                                   check=True)
-            proc_lines = proc.stdout.decode().splitlines()
+            configs_json = json.loads(proc.stdout.decode())
+            for config in configs_json['configurations']:
+                if cfs_config_prefix in config['name']:
+                    found_configurations.append(config['name'])
 
-            filtered_lines = [line for line in proc_lines if cfs_config_prefix in line]
+        except subprocess.CalledProcessError as err:
+            logging.warning('Failed to find CFS configurations with prefix "%s" '
+                            'created by test: %s', cfs_config_prefix, err.stderr)
 
-            simplified_values = [re.search(r'"([^"]*)"$', item).group(1) for item in filtered_lines]
+        for configuration_name in found_configurations:
+            cls.delete_cfs_configuration(configuration_name)
 
-            found_configurations = simplified_values
+    @classmethod
+    def delete_all_ims_images_matching_prefix(cls, cfs_config_prefix):
+        """Find and delete all IMS images matching a prefix using the 'cray' CLI.
+
+        This relies on the cray CLI being configured and authenticated on the system.
+
+        Args:
+            cfs_config_prefix (str): ims image prefix to match
+        """
+        find_command = 'cray ims images list'
+        found_image_ids = []
+        try:
+            proc = subprocess.run(shlex.split(find_command), stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                                  check=True)
+            images_json = json.loads(proc.stdout.decode())
+            for image in images_json:
+                if cfs_config_prefix in image['name']:
+                    found_image_ids.append(image['id'])
 
         except subprocess.CalledProcessError as err:
             logging.warning('Failed to find CFS configurations with prefix "%s" '
                             'created by test: %s', cfs_config_prefix, err.stderr)
 
-        for configuration_name in found_configurations:
-            delete_command = f'cray cfs configurations delete {configuration_name}'
-            try:
-                subprocess.run(shlex.split(delete_command), check=True)
-            except subprocess.CalledProcessError as err:
-                logging.warning('Failed to delete CFS configuration "%s" '
-                                'created by test: %s', configuration_name, err.stderr)
+        for image_id in found_image_ids:
+            cls.delete_ims_image(image_id)
+
+    @classmethod
+    def delete_all_session_templates_matching_prefix(cls, session_template_prefix):
+        """Find and delete all BOS session templates matching a prefix using the 'cray' CLI.
+
+        This relies on the cray CLI being configured and authenticated on the system.
+
+        Args:
+            session_template_prefix (str): session template prefix to match
+        """
+        find_command = 'cray bos sessiontemplates list'
```

Comment:
We're not consistent about this, but we should specify the API version in our commands for protection against changes in the commands if a new version comes out (admittedly unlikely at this point).

```suggestion
        find_command = 'cray bos v2 sessiontemplates list'
```